### PR TITLE
Radio binder take 2

### PR DESCRIPTION
**Pull request by vlymar**
This is my second attempt at fixing https://github.com/dobtco/screendoor-v2/issues/4557 with the least amount of hackery.

Its an alternative to the already merged previous attempt at a fix: https://github.com/dobtco/formrenderer-base/pull/143, and a slight improvement over my first attempt (never merged) https://github.com/dobtco/formrenderer-base/pull/148.

The radio view no longer shares a template with the checkboxes. Part of what made all of this confusing was the shared logic between checkboxes and radios.

Additionally, radio inputs no longer use `checked` or `checkedarray` binders. They now use a `dobtradiogroup` binder. Also note that the "other" radio input is bound to the same model value as the rest (`model.value.checked`). Having 'other' bound to a different model value caused the inputs to be in two separate logical groups, and introduced the need for hacks to deselect inputs via jquery. I wanted to fix this in the simplest possible way, so I created a new binder that simply checks if the selection event happened on the 'other' input (identified by a special class found in the template). If the event happens on the 'other' input, it sets `other_checked` to true.

Its also necessary to manually unset `other_checked` and `other_text` when the event is fired on a non-other field, because nothing in rivets will do that.

At this point I consider rivets a source of tech debt. Its a dependency we haven't updated in forever, so even getting up to date with it would take forever, but also, its not a very easy library to use and is no longer maintained. I don't think struggling to find a 'clean' way to make radio groups work within the much greater mess of this situation is really worth it.

This PR very slightly modifies the payload sent. `other_checked` now sends `true` instead of `"on"`. I will test to make sure this is ok.

This can be tested locally by pulling this branch down, then going to your screendoor repo and running `script/fr-copy copy /path/to/formrenderer-rbase

**TODO**

- [x] Rebuild dist/
- [x] Test with Screendoor local
- [x] Code review
- [x] Test on staging
- [x] Tag
- [x] Release

### CODE REVIEW

File: src/templates/fields/checkboxes.eco [hunk 1]
```diff
@@ -1 +1,20 @@
-<%- JST["partials/options_field"](@) %>
+<% for option in @model.getOptions(): %>
```
Comment (vlymar) [inline src/templates/fields/checkboxes.eco]: This is an exact copy of what options_field.eco used to be

File: src/templates/fields/radio.eco [hunk 2]
```diff
@@ -1 +1,27 @@
-<%- JST["partials/options_field"](@) %>
+<% for option in @model.getOptions(): %>
```
Comment (vlymar) [inline src/templates/fields/radio.eco]: this has been changed from options_field.eco to use the new `dobtradiogroup` binder.

**Comment by vlymar** [comment]
I've tested this locally with Screendoor and everything appears to work right.

Will release a release candidate for testing on staging after code review.

**Comment by vlymar** [comment]
side note - we may want to modify `script/release` to allow pushing custom branches.

**Comment by kariannemah** [comment]
Any chance we can get this on staging in the next day or two? I'm eager to test whether it fixes the bug that ProPublica reported in https://app.frontapp.com/open/cnv_7g2bvl.

**Comment by vlymar** [comment]
working on it right now

**Comment by vlymar** [comment]
ok @kariannemah, its in staging

**Comment by vlymar** [comment]
I see a new issue I've never noticed before though.

screen shot 2017-09-19 at 1 33 03 pm

If you don't select the 'Other' field, when you go to preview you won't see the other option at all.

This includes in the response admin view and on exported PDFS.

screen shot 2017-09-19 at 1 34 27 pm
screen shot 2017-09-19 at 1 34 35 pm

As far as I can tell this is happening in master too.

Is this a known bug?

**Comment by vlymar** [comment]
@kariannemah do you have the bandwidth to do a quick QA pass on this in staging? If not I can poke around real quick.

 I'm reasonably confident that this is ready for deploy.

**Comment by kariannemah** [comment]
Yup, let me take a quick look right now.

**Comment by kariannemah** [comment]
I got a new phone so 2FA doesn't work for me on staging anymore. 😞 

@vlymar, go ahead with a quick QA on staging. I'll wait until the deploy to test the ProPublica bug.

**Comment by kariannemah** [comment]
I'm able to replicate https://github.com/dobtco/formrenderer-base/pull/149#issuecomment-330665438 in production. Never noticed it before. We probably neglected to test this case before https://github.com/dobtco/screendoor-v2/pull/2270 was merged.

**Comment by vlymar** [comment]
Ok, it looks good to me. will be deploying soon.

Will open a new ticket for the other issue

**Comment by kariannemah** [comment]
Victor fixed my 2FA on staging, so I was able to test this. Looks solid to me! 👍

**Comment by vlymar** [comment]
Filed https://github.com/dobtco/screendoor-v2/issues/4716 to for the 'other' field not showing up

**Pull request merged**